ports/esp32/network_ppp.c

```diff
@@ -115,61 +115,124 @@ static void pppos_client_task(void *self_in) {
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_self, MP_ARG_OBJ | MP_ARG_REQUIRED, {.u_obj = MP_OBJ_NULL} },
+        { MP_QSTR_auth_method, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL} },
+        { MP_QSTR_user_name, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL} },
+        { MP_QSTR_password, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL} },
+    };
 
-            self->pcb = pppapi_pppos_create(&self->pppif, ppp_output_callback, ppp_status_cb, self);
+    mp_arg_val_t parsed_args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args, args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, parsed_args);
 
-            if (self->pcb == NULL) {
-                mp_raise_msg(&mp_type_RuntimeError, "init failed");
-            }
-            pppapi_set_default(self->pcb);
-            ppp_set_usepeerdns(self->pcb, 1);
-            pppapi_connect(self->pcb, 0);
+    ppp_if_obj_t *self = MP_OBJ_TO_PTR(parsed_args[ARG_self].u_obj);
 
-            xTaskCreatePinnedToCore(pppos_client_task, "ppp", 2048, self, 1, (TaskHandle_t*)&self->client_task_handle, MP_TASK_COREID);
-            self->active = true;
-        } else {
-            if (!self->active) {
-                return mp_const_false;
-            }
-
-            // Wait for PPPERR_USER, with timeout
-            pppapi_close(self->pcb, 0);
-            uint32_t t0 = mp_hal_ticks_ms();
-            while (!self->clean_close && mp_hal_ticks_ms() - t0 < PPP_CLOSE_TIMEOUT_MS) {
-                mp_hal_delay_ms(10);
-            }
-
-            // Shutdown task
-            xTaskNotifyGive(self->client_task_handle);
-            t0 = mp_hal_ticks_ms();
-            while (self->client_task_handle != NULL && mp_hal_ticks_ms() - t0 < PPP_CLOSE_TIMEOUT_MS) {
-                mp_hal_delay_ms(10);
-            }
-
-            // Release PPP
-            pppapi_free(self->pcb);
-            self->pcb = NULL;
-            self->active = false;
-            self->connected = false;
-            self->clean_close = false;
+    mp_obj_t auth_method = parsed_args[ARG_auth_method].u_obj;
+    mp_obj_t user_name   = parsed_args[ARG_user_name].u_obj;
+    mp_obj_t password    = parsed_args[ARG_password].u_obj;
+    uint8_t auth_type = PPPAUTHTYPE_NONE;
+    if ( auth_method != MP_OBJ_NULL ) {
+        if ( !mp_obj_is_str(auth_method) ) {
+            mp_raise_TypeError("auth_method must be a str");
+        }
+        if ( user_name == MP_OBJ_NULL ) {
+            mp_raise_ValueError("user name must not be None");
         }
+        if ( !mp_obj_is_str(user_name) ) {
+            mp_raise_TypeError("user name must be a str");
+        }
+        if ( password == MP_OBJ_NULL ) {
+            mp_raise_ValueError("password must not be None");
+        }
+        if ( !mp_obj_is_str(password) ) {
+            mp_raise_TypeError("password must be a str");
+        }
+        qstr auth_method_qstr = mp_obj_str_get_qstr(auth_method);
```

Comment:
Instead of a string, can use constants (see below).

Reply:
Changed to use integer constants.